### PR TITLE
FXO digitals pricer

### PR DESCRIPTION
**Pull request by Rick4020**
FXO digitals pricer and testing scripts

### CODE REVIEW

File: modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxDigitalOption.java [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
+ * A vanilla Digital FX option.
+ * <p>
+ * An FX option is a financial instrument that provides an option based on the future value of
+ * a foreign exchange. The option is European, exercised only on the exercise date.
+ * <p>
+ * For example, a call on a 'EUR 1.00 / USD -1.41' exchange is the option to
+ * perform a foreign exchange on the expiry date, where USD 1.41 is paid to receive EUR 1.00.
+ */
+@BeanDefinition
+public final class FxDigitalOption
```
Comment (andreiruse) [inline modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxDigitalOption.java]: I couldn't find a unit test which covers this class - can you please add one?

File: modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxDigitalOptionTrade.java [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+/**
+ * A trade in a vanilla FX option.
+ * <p>
+ * An Over-The-Counter (OTC) trade in an {@link FxDigitalOption}.
+ * <p>
+ * An FX Digital option is a financial instrument that provides a binary payoff
+ * The option is European, exercised only on the exercise date.
+*/
+@BeanDefinition
+public final class FxDigitalOptionTrade
```
Comment (andreiruse) [inline modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxDigitalOptionTrade.java]: I couldn't find a unit test which covers this class - could you please add one?

File: modules/product/src/main/java/com/opengamma/strata/product/fxopt/ResolvedFxDigitalOptionTrade.java [hunk 3]
```diff
@@ -0,0 +1,474 @@
...
+ * This is the resolved form of {@link FxDigitalOptionTrade} and is the primary input to the pricers.
+ * Applications will typically create a {@code ResolvedFxDigitalOptionTrade} from a {@code FxDigitalOptionTrade}
+ * using {@link FxDigitalOptionTrade#resolve(ReferenceData)}.
+ * <p>
+ * A {@code ResolvedFxDigitalOptionTrade} is bound to data that changes over time, such as holiday calendars.
+ * If the data changes, such as the addition of a new holiday, the resolved form will not be updated.
+ * Care must be taken when placing the resolved form in a cache or persistence layer.
+ */
+@BeanDefinition
+public final class ResolvedFxDigitalOptionTrade
```
Comment (andreiruse) [inline modules/product/src/main/java/com/opengamma/strata/product/fxopt/ResolvedFxDigitalOptionTrade.java]: I couldn't find a unit test which covers this class - could you please add one?

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/fxopt/BlackFxDigitalOptionTradePricer.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+import com.opengamma.strata.product.fxopt.ResolvedFxSingleBarrierOption;
+import com.opengamma.strata.product.fxopt.ResolvedFxDigitalOption;
+import com.opengamma.strata.product.fxopt.ResolvedFxDigitalOptionTrade;
+
+/**
+ * Pricer for FX vanilla option trades with a lognormal model.
+ * <p>
+ * This function provides the ability to price an {@link FxDigitalOptionTrade}.
+ */
+public class BlackFxDigitalOptionTradePricer {
```
Comment (andreiruse) [inline modules/pricer/src/main/java/com/opengamma/strata/pricer/fxopt/BlackFxDigitalOptionTradePricer.java]: I couldn't find a unit test which covers this class - can you please add one?

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/fxopt/BlackFxDigitalOptionTradePricer.java [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+     * Calculates the present value of the FX vanilla option trade.
+     * <p>
+     * The present value of the trade is the value on the valuation date.
+     *
+     * @param trade  the option trade
+     * @param ratesProvider  the rates provider
+     * @param volatilities  the Black volatility provider
+     * @return the present value of the trade
+     */
+    public MultiCurrencyAmount presentValue(
```
Comment (jodastephen) [inline modules/pricer/src/main/java/com/opengamma/strata/pricer/fxopt/BlackFxDigitalOptionTradePricer.java]: These names are consistent with the rest of the pricer classes ;-)

File: modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxDigitalOption.java [hunk 6]
```diff
@@ -0,0 +1,641 @@
...
+/**
+ * A vanilla Digital FX option.
+ * <p>
+ * An FX option is a financial instrument that provides an option based on the future value of
+ * a foreign exchange. The option is European, exercised only on the exercise date.
+ * <p>
+ * For example, a call on a 'EUR 1.00 / USD -1.41' exchange is the option to
+ * perform a foreign exchange on the expiry date, where USD 1.41 is paid to receive EUR 1.00.
+ */
+@BeanDefinition
```
Comment (jodastephen) [inline modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxDigitalOption.java]: As it stands, all beans in the trade model should be generated as full beans, not light or minimal. If appropriate, the builder can be made private using builderStyle="private", but I don't think that applies here.

**Comment by marc-henrard** [comment]
I guess that the classes 'SwapPricingExample', 'FxRate', 'FxVanillaOptionTest', 'BlackFxVanillaOptionProductPricerTest', 'BlackOneTouchAssetPriceFormulaRepository' where modified my mistake. Revert to the previous version.

**Comment by Rick4020** [comment]
Yes i will revert those classes.

**Comment by andreiruse** [comment]
As discussed with @marc-henrard, closing the PR for now - will revisit the branch later.

**Pull request closed without merging**